JoinColony/colonyCDapp
# Fix user hub stakes tab advanced payments

## Description

- Ensure stakes made for an advanced payment have all details shown properly in the user hub stakes tab

## Testing

* Ensure you have the Staking advanced payments extension enabled.
* Create an advanced payment (Payment builder) action. You can use any number of payments / users / tokens.
* Use the staking decision method to create it.
<img width="1075" alt="Screenshot 2024-07-15 at 17 55 18" src="https://github.com/user-attachments/assets/36a6ffdb-419e-47d2-a83b-dfe00d2e6578">

* Check the action details in the Userhub Stakes tab. The name, metadata description and status should all be correct.
<img width="1075" alt="Screenshot 2024-07-15 at 17 55 46" src="https://github.com/user-attachments/assets/19f28441-f23b-4efa-ba2c-f7d7a106086e">

* Go through all the steps in the action sidebar to complete the action fully and reclaim your stake.
<img width="1075" alt="Screenshot 2024-07-15 at 17 56 41" src="https://github.com/user-attachments/assets/b2d2a6d1-6813-47b3-80a8-9371310bb6cd">

* The status in the stakes tab should be updated.
<img width="1075" alt="Screenshot 2024-07-15 at 17 56 19" src="https://github.com/user-attachments/assets/20815257-fcb4-4938-8955-5fbabdd84456">


## Diffs

**Changes** 🏗

* Ensure action metadata is created when a staked expenditure is made
* Load and display expenditure data when needed in the stakes tab of the user hub
* Ensure the correct user stake status is shown for expenditures in the stakes tab of the user hub

Resolves #2536
